### DISCUSSION TITLE
AWS::S3::Errors::Forbidden - Ruby on Rails

### DISCUSSION THREAD

**Discussion by AdamTelos** [aws/aws-sdk-ruby#3124] [Q&A]
Hi,

I'm running a old Rails app. Have been getting this error coinciding with the release of [1.167.0](https://rubygems.org/gems/aws-sdk-s3/versions/1.167.0) - October 02, 2024

I am using:
gem 'aws-sdk' 
Inside of my gemfile

I have tried including:
gem 'aws-sdk-s3', '~> 1.166'

Below this, as this is what updated on Oct-2 and may be what's causing this issue. (I did not deploy any new code that day). However I think there might be something new inside of aws-sdk that's clashing as I am still getting the same problem.

Any advice on how to get my gem 'aws-sdk' to a pre Oct-2 state? It's causing issues at my company.

**Comment by mullermp** [comment]
Hello. The error `AWS::S3::Errors::Forbidden` (particularly the namespace AWS) indicates you may be using version 1 of the Ruby SDK? Adding `aws-sdk-s3` would not solve any issue unless you make code changes. I would not recommend using version 1 and I would also not recommend using the `aws-sdk` gem and instead use the modular gem such as `aws-sdk-s3` as you tried adding. Forbidden may be an issue of what credentials you have configured. I think more information about your code and configuration is needed.
